tests/utils.py

```diff
@@ -220,3 +249,247 @@ def dict_without(input_dict: Mapping[K, V], without_keys: set[str]) -> dict[K, V
     {'foo': 'bar', 'bar': 'baz', 'zip': 'zap'}
     """
     return {k: v for k, v in input_dict.items() if k not in without_keys}
+
+
+T_Object = TypeVar("T_Object", bound=object)
+
+
+class FakeCogniteResourceGenerator:
+    _error_msg: typing.ClassVar[str] = "Please extend this function to support generating fake data for this type"
```

Comment:
This is quite the beast 😉 I wonder how easy it will be for someone to extend it :hide:

Reply:
Yes, that is an issue. However, it is useful to have it as it detects so many bugs. I expect I will be bothered if this line is hit.